### PR TITLE
Support sofa registry

### PR DESCRIPTION
**Pull request by leizhiyuan**
### Motivation:

Explain the context, and why you're making that change.
To make others understand what is the problem you're trying to solve.

### Modification:

Describe the idea and modifications you've done.

### Result:

Fixes #<GitHub issue number>. 

If there is no issue then describe the changes introduced by this PR.

### CODE REVIEW

**Comment by leizhiyuan** [comment]
https://github.com/alipay/sofa-rpc/issues/552

**Comment by leizhiyuan** [comment]
准备将reigstry config回调先去掉。只订阅地址，之后模型重构完再添加

**Comment by leizhiyuan** [comment]
test cases in jdk7 will fail， we can ignore it

File: extension-impl/extension-common/src/main/java/com/alipay/sofa/rpc/common/SofaRegistryConstants.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ * limitations under the License.
+ */
+package com.alipay.sofa.rpc.common;
+
+/**
+ * Created by zhanggeng on 2017/7/5.
+ *
+ * @author <a href="mailto:zhanggeng.zg@antfin.com">zhanggeng</a>
+ */
+public class SofaRegistryConstants {
```
Comment (ujjboy) [inline extension-impl/extension-common/src/main/java/com/alipay/sofa/rpc/common/SofaRegistryConstants.java]: 这个类就 SOFARegistry 自己模块用的吧。不用放到 common 下

Comment (leizhiyuan) [reply extension-impl/extension-common/src/main/java/com/alipay/sofa/rpc/common/SofaRegistryConstants.java]: 好的。

**Comment by leizhiyuan** [comment]
please review this change. thanks~

File: bom/pom.xml [hunk 2]
```diff
@@ -23,6 +23,7 @@
         <opentracing.version>0.22.0</opentracing.version>
         <dubbo.version>2.4.10</dubbo.version>
         <nacos.version>0.6.0</nacos.version>
+        <sofa.registry.version>5.2.0-SNAPSHOT</sofa.registry.version>
```
Comment (JervyShi) [inline bom/pom.xml]: 可以切换为 5.2.0 正式版了

Comment (leizhiyuan) [reply bom/pom.xml]: 好

File: core/common/src/main/java/com/alipay/sofa/rpc/common/utils/StringUtils.java [hunk 3]
```diff
@@ -416,4 +416,45 @@ public static String substringAfter(String str, String separator) {
...
+     * StringUtils.substringBefore("abc", null)  = "abc"
+     * </pre>
+     *
+     * @param str  the String to get a substring from, may be null
+     * @param separator  the String to search for, may be null
+     * @return the substring before the first occurrence of the separator,
+     *  <code>null</code> if null String input
+     * @since 2.0
+     */
+    public static String substringBefore(String str, String separator) {
```
Comment (ujjboy) [inline core/common/src/main/java/com/alipay/sofa/rpc/common/utils/StringUtils.java]: add test case for this method.

**Pull request merged**